src/app/api/run/[runId]/approve/route.test.ts

```diff
@@ -0,0 +1,50 @@
+import { POST } from '@/app/api/run/[runId]/approve/route'
+import { afterEach, beforeEach, describe, expect, it } from 'vitest'
+import path from 'path'
+import { UPLOAD_DIR } from '@/app/utils'
+import fs from 'fs'
+
+describe('POST /api/run/:runId/approve', () => {
+    const mockFileContent = 'header1,header2\nvalue1,value2'
+    const runId = 'test-run-id'
+    const mockFilePath: string = path.join(UPLOAD_DIR, runId)
+
+    // Create a temporary test directory before each test
+    beforeEach(() => {
+        // recreate directory
+        fs.mkdirSync(UPLOAD_DIR, { recursive: true })
```

Comment:
I've used [mock-fs](https://github.com/tschaub/mock-fs) with pretty good results for filesystem-based tests

Reply:
i will incorporate mock-fs! thanks for the suggestion. i did come across it when first looking at things